src/khoj/processor/content/images/image_to_entries.py

```diff
@@ -0,0 +1,124 @@
+        with timer("Split entries by max token size supported by model", logger):
+            current_entries = self.split_entries_by_max_tokens(current_entries, max_tokens=256)
+
+        # Identify, mark and merge any new entries with previous entries
+        with timer("Identify new or updated entries", logger):
+            num_new_embeddings, num_deleted_embeddings = self.update_embeddings(
+                current_entries,
+                DbEntry.EntryType.IMAGE,
+                DbEntry.EntrySource.COMPUTER,
+                "compiled",
+                logger,
+                deletion_file_names,
+                user,
+                regenerate=regenerate,
+                file_to_text_map=file_to_text_map,
+            )
+
+        return num_new_embeddings, num_deleted_embeddings
+
+    @staticmethod
+    def extract_image_entries(image_files) -> Tuple[Dict, List[Entry]]:  # important function
+        """Extract entries by page from specified image files"""
+        file_to_text_map = dict()
+        entries: List[str] = []
+        entry_to_location_map: List[Tuple[str, str]] = []
+        for image_file in image_files:
+            try:
+                try:
+                    loader = RapidOCR()
+                    bytes = image_files[image_file]
+                    # write the image to a temporary file
+                    timestamp_now = datetime.utcnow().timestamp()
+                    timestamp_now = datetime.utcnow().timestamp()
+                    # use either png or jpg
+                    if image_file.endswith(".png"):
+                        tmp_file = f"tmp_image_file_{timestamp_now}.png"
+                    elif image_file.endswith(".jpg") or image_file.endswith(".jpeg"):
+                        tmp_file = f"tmp_image_file_{timestamp_now}.jpg"
+                    with open(f"{tmp_file}", "wb") as f:
+                        bytes = image_files[image_file]
+                        f.write(bytes)
+                    try:
+                        image_entries_per_file = []
+                        result, _ = loader(tmp_file)
+                        if result:
+                            image_entries_per_file = [text[1] for text in result]
+                    except ImportError:
+                        logger.warning(f"Unable to process file: {image_file}. This file will not be indexed.")
+                        continue
+                except ImportError:
+                    logger.warning(f"Unable to process file: {image_file}. This file will not be indexed.")
+                    continue
+                entry_to_location_map += zip(
+                    image_entries_per_file, [image_file] * len(image_entries_per_file)
+                )  # this is an indexed map of image_entries for the pdf.
+                entries.extend(image_entries_per_file)
+                file_to_text_map[image_file] = image_entries_per_file
+            except Exception as e:
```

Comment:
Are so many layers of try/catch required? Do we expect the code running in the outermost try/catch to fail under normal operating conditions? If not, we should continue to raise exceptions for those to make unexpected failures in this code path apparent

Reply:
good point! fixed.